src/script/lua_api/l_object.cpp

```diff
@@ -895,6 +895,10 @@ int ObjectRef::l_set_yaw(lua_State *L)
 	ObjectRef *ref = checkobject(L, 1);
 	LuaEntitySAO *co = getluaobject(ref);
 	if (co == NULL) return 0;
+	if (isNaN(L, 2)) {
+		throw LuaError("ObjectRef::set_yaw: NaN value is not allowed.");
```

Comment:
Needs backtrace to the caller mod. Brace removal possible.
How about defining `CHECK_FLOAT_NAN`, similar to what's used in `c_content.cpp` to verify the float values?

Reply:
@SmallJoker i think a float generic reading function is better to ensure this is fixed on the whole code

Reply:
i don't find any way to find it currently in the code, all LuaErrors are thrown like this

Reply:
I know that we sometimes throw LuaErrors. But including a backtrace like found in `log_deprecated` would be helpful to find out what mod supplied unacceptable values:
https://github.com/minetest/minetest/blob/df991edaa8617730f4ad15648e9131325a0fe2e8/src/script/common/c_internal.cpp#L154-L161
Since this function is always called directly from the mods, we could use this simple code:
```C++
lua_Debug ar;

-- Direct call from the mod, check stack 1 directly
FATAL_ERROR_IF(!lua_getstack(L, 1, &ar), "lua_getstack() failed");
FATAL_ERROR_IF(!lua_getinfo(L, "Sl", &ar), "lua_getinfo() failed");
warningstream << " (at " << ar.short_src << ":" << ar.currentline << ")";
```